pkg/operator/encryption/controllers/condition_controller.go

```diff
@@ -0,0 +1,242 @@
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/wait"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+
+	"github.com/openshift/library-go/pkg/operator/encryption/encryptionconfig"
+	"github.com/openshift/library-go/pkg/operator/encryption/state"
+	"github.com/openshift/library-go/pkg/operator/encryption/statemachine"
+	"github.com/openshift/library-go/pkg/operator/events"
+	operatorv1helpers "github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+const (
+	conditionWorkKey = "key"
+)
+
+// conditionController maintains the Encrypted condition. It sets it to true iff there is a
+// fully migrated read-key in the current config, and no later key is of identity type.
+type conditionController struct {
+	operatorClient operatorv1helpers.OperatorClient
+
+	queue         workqueue.RateLimitingInterface
+	eventRecorder events.Recorder
+
+	preRunCachesSynced []cache.InformerSynced
+
+	encryptedGRs []schema.GroupResource
+
+	encryptionSecretSelector metav1.ListOptions
+
+	deployer     statemachine.Deployer
+	secretClient corev1client.SecretsGetter
+}
+
+func NewConditionController(
+	deployer statemachine.Deployer,
+	operatorClient operatorv1helpers.OperatorClient,
+	kubeInformersForNamespaces operatorv1helpers.KubeInformersForNamespaces,
+	secretClient corev1client.SecretsGetter,
+	encryptionSecretSelector metav1.ListOptions,
+	eventRecorder events.Recorder,
+	encryptedGRs []schema.GroupResource,
+) *conditionController {
+	c := &conditionController{
+		operatorClient: operatorClient,
+
+		queue:         workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "EncryptionConditionController"),
+		eventRecorder: eventRecorder.WithComponentSuffix("encryption-condition-controller"),
+
+		encryptedGRs: encryptedGRs,
+
+		encryptionSecretSelector: encryptionSecretSelector,
+		deployer:                 deployer,
+		secretClient:             secretClient,
+	}
+
+	c.preRunCachesSynced = setUpInformers(deployer, operatorClient, kubeInformersForNamespaces, c.eventHandler())
+
+	return c
+}
+
+func (c *conditionController) sync() error {
+	if ready, err := shouldRunEncryptionController(c.operatorClient); err != nil || !ready {
+		return err // we will get re-kicked when the operator status updates
+	}
+
+	currentConfig, desiredState, foundSecrets, transitioningReason, err := statemachine.GetEncryptionConfigAndState(c.deployer, c.secretClient, c.encryptionSecretSelector, c.encryptedGRs)
+	if err != nil || len(transitioningReason) > 0 {
+		return err
+	}
+
+	cond := operatorv1.OperatorCondition{
+		Type:    "Encrypted",
+		Status:  operatorv1.ConditionTrue,
+		Reason:  "EncryptionComplete",
+		Message: fmt.Sprintf("Resources all encrypted: %s", grString(c.encryptedGRs)),
+	}
+	currentState := encryptionconfig.ToEncryptionState(currentConfig)
```

Comment:
we need to update it wrt https://github.com/openshift/library-go/pull/612

Reply:
done